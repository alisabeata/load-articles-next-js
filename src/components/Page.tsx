'use client'

import React, { useState, useCallback } from 'react'
import { Button } from '@/components/Button'
import { DataView, DataType } from '@/components/DataView'

const Page: React.FC = () => {
  const [data, setData] = useState<DataType>(null)
  const [isLoaded, setIsLoaded] = useState(false)

  const fetchData = async () => {
    try {
      // Load the data using fetch
      const response = await fetch('https://jsonplaceholder.typicode.com/posts')
      if (!response.ok) {
        // Error handler
        throw new Error('Failed to fetch data')
      }
      const jsonData = await response.json()
      setData(jsonData)
      setIsLoaded(true)
    } catch (error) {
      console.warn('Failed to fetch data')
    }
  }

  return (
    <>
      <Button onClick={fetchData} isLoaded={isLoaded} setIsLoaded={setIsLoaded}>
        Load Data
      </Button>
      <DataView data={data} />
    </>
  )
}

export default Page
