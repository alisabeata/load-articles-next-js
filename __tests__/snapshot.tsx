/**
 * @jest-environment jsdom
 */
import { render } from '@testing-library/react'
import Page from '@/app/page'

it('renders Page', () => {
  const { container } = render(<Page />)
  expect(container).toMatchSnapshot()
})
