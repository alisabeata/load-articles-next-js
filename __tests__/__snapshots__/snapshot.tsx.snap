// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Page 1`] = `
<div>
  <button>
    Load Data
  </button>
</div>
`;
